source/configuration.txt

```diff
@@ -1,16 +1,160 @@
-TODO: write
+.. _visual-studio-extension-configuration:
 
-TODO: Incorporate Telemetery Information
+==============================
+Configure the {+product+}
+==============================
 
-.. important:: Telemetry
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Learn how to configure the {+product+} in your .NET project and what
+configuration options are available.
+
+Configuration
+-------------
+
+To configure the {+product+}, perform the following actions:
+
+#. Install the {+product+} as described in the :doc:`Install </install>` guide.
+
+#. Create a file specifying your configuration for the {+product+} named
+   ``{+product-settings-file-constant+}``.
    
-   When the {+product+} analyzes your source code, the extension
-   sends the following information to MongoDB:
+   .. important:: Naming
+
+      You can put your settings file in any directory, but you must name your
+      file ``{+product-settings-file-constant+}``.
+
+#. Add your configuration to your ``{+product-settings-file-constant+}`` file.
+
+   The following sample configuration makes the {+product+} use the 
+   `LINQ3 <{+driver-docs+}reference/driver/crud/linq3/>`__
+   provider.
+
+   .. code-block:: json
+
+      { "DefaultLinqVersion": "V3" }
+
+#. Add the following to the ``.csproj`` for your .NET application within the ``Project`` tag:
+
+   .. code-block:: xml
+
+      <ItemGroup>
+        <AdditionalFiles Include="<path to your {+product-settings-file-constant+} file>" />
+      </ItemGroup>
+
+To learn more about .NET project files, see 
+`Understanding the Project File <https://docs.microsoft.com/en-us/aspnet/web-forms/overview/deployment/web-deployment-in-the-enterprise/understanding-the-project-file>`__
+from Microsoft.
+
+To learn more about the ``AdditionalFiles`` tag, see 
+:github:`Using Additional Files </dotnet/roslyn/blob/main/docs/analyzers/Using%20Additional%20Files.md>`
+in the Roslyn documentation.
+
+Options
+-------
+
+The following is a description of all configuration options supported
+by the {+product+}:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 35 65
+
+   * - Name
+     - Description
+
+   * - | **DefaultLinqVersion**
+     - | **Type:** string
```

Comment:
question: Is type necessary since we're showing what the default values are?

Reply:
I got this [style of options from the Kafka Docs](https://docs.mongodb.com/kafka-connector/current/sink-connector/configuration-properties/mongodb-connection/), and I thought it was helpful there (I would rather explicitly state the type than have people potentially unfamiliar with this table's format infer it).

The only semi-concrete reason to keep type information here is to distinguish between boolean `true` and the string `"true"`.

I'm a fan of keeping the types (I think it might help someone understand it) but there is definitely an argument to remove them. If there is a strong opinion to remove them I can remove them, but I have a slight preference for keeping them.

Reply:
I'd measure whether any perceived value gained correlates with measurable feedback from users. I'll leave it up to you.